core/src/cubos/core/data/ser/json.cpp

```diff
@@ -0,0 +1,128 @@
+#include <cubos/core/log.hpp>
+
+/// [JSON serializer]
+#include <cubos/core/data/ser/json.hpp>
+
+using cubos::core::data::JSONSerializer;
+using cubos::core::reflection::Type;
+
+/// [MACRO to add hooks for primitive types]
```

Comment:
```suggestion
// Macro used to reduce code duplication for primitive type hooks.
```
I think you misunderstood those /// [] comments: they're meant to be used only in samples, so that we can reference them in the page itself. In ordinary source files, you can just use normal comments formatting.